### PR TITLE
Get emoji from reaction

### PR DESCRIPTION
**Pull request by GrapeColor**
# Summary
Add emoji attribute to Reaction to get reaction Emoji.
Integrate the fixes in #689.
Fork from branch #671 to avoid conflicts.

## Added
Added emoji attribute to Reaction.
Added server argument to `Reaction#initialize`.

## Changed
Changed the return value for Unicode emoji in `Emoji#mention`.
Changed to set server argument basically when initializing Emoji object.
Changed `Reaction#to_s` to call `Emoji#to_reaction`.

## Fixed
Fixed an argument setting error in parsing emoji in `Bot#parse_mentions`.

## P.S.
I can't write English, so I use a translator.
I'm sorry...

### CODE REVIEW

**Comment by GrapeColor** [comment]
There is no file for running circleci on this branch, so it seems to be "All checks have failed". 
What should I do?

**Comment by swarley** [comment]
I've just updated my PR. If you rebase again and push you should be able to trigger the CI

**Comment by GrapeColor** [comment]
Thank you for presenting the solution for CI.
It was helpful.

File: lib/discordrb/data/emoji.rb [hunk 1]
```diff
@@ -42,17 +42,15 @@ def ==(other)
 
     # @return [String] the layout to mention it (or have it used) in a message
     def mention
-      "<#{'a' if animated}:#{name}:#{id}>"
+      id.nil? ? name : "<#{'a' if animated}:#{name}:#{id}>"
```
Comment (GrapeColor) [inline lib/discordrb/data/emoji.rb:45]: Should it be fixed as it conflicts with #715?

**Comment by Daniel-Worrall** [comment]
The non-breaking changes were merged (with changes) via shardlab/discordrb@18c7e71269fe09b526623f68d7d200a6ff3159f8 in the [new repo](https://github.com/shardlab/discordrb) and released in [v3.4.0](https://github.com/shardlab/discordrb/releases/tag/v3.4.0). We intend to discuss changes (such as breaking changes) to Emoji/Reaction for better support/developer experience for 4.0

**Pull request closed without merging**